samples/js-tests/src/XHRTest/XHRTest.js

```diff
@@ -57,96 +84,75 @@ var XHRTestLayer = cc.Layer.extend({
-
-        xhr.onreadystatechange = function () {
-            if (xhr.readyState == 4 && (xhr.status >= 200 && xhr.status <= 207)) {
-                var httpStatus = xhr.statusText;
-                var response = xhr.responseText.substring(0, 100) + "...";
-                var responseLabel = new cc.LabelTTF("GET Response (100 chars): \n" + response, "Thonburi", 16);
-                that.addChild(responseLabel, 1);
-                responseLabel.anchorX = 0;
-                responseLabel.anchorY = 1;
-                responseLabel.textAlign = cc.TEXT_ALIGNMENT_LEFT;
-
-                responseLabel.x = 10;
-                responseLabel.y = winSize.height / 2;
-                statusGetLabel.setString("Status: Got GET response! " + httpStatus);
-            }
-        };
         xhr.send();
+
+        streamXHREventsToLabel(xhr, statusGetLabel, responseLabel, "GET");
     },
 
     sendPostPlainText: function() {
-        var that = this;
-        var xhr = cc.loader.getXMLHttpRequest();
-        var statusPostLabel = new cc.LabelTTF("Status:", "Thonburi", 18);
+        var statusPostLabel = new cc.LabelTTF("Status:", "Thonburi", 12);
         this.addChild(statusPostLabel, 1);
 
-        statusPostLabel.x = winSize.width / 2;
-
-        statusPostLabel.y = winSize.height - 140;
+        statusPostLabel.x = winSize.width / 10 * 3;
+        statusPostLabel.y = winSize.height - 100;
+        ensureLeftAligned(statusPostLabel);
         statusPostLabel.setString("Status: Send Post Request to httpbin.org with plain text");
 
+
+        var responseLabel = new cc.LabelTTF("", "Thonburi", 16);
+        this.addChild(responseLabel, 1);
+        ensureLeftAligned(responseLabel);
+        responseLabel.x = winSize.width / 10 * 3;
+        responseLabel.y = winSize.height / 2;
+        
+        var xhr = cc.loader.getXMLHttpRequest();
         xhr.open("POST", "http://httpbin.org/post");
         //set Content-type "text/plain;charset=UTF-8" to post plain text
         xhr.setRequestHeader("Content-Type","text/plain;charset=UTF-8");
-        xhr.onreadystatechange = function () {
-            if (xhr.readyState == 4 && (xhr.status >= 200 && xhr.status <= 207)) {
-                var httpStatus = xhr.statusText;
-                var response = xhr.responseText.substring(0, 100) + "...";
-                var responseLabel = new cc.LabelTTF("POST Response (100 chars):  \n" + response, "Thonburi", 16);
-                that.addChild(responseLabel, 1);
-                responseLabel.anchorX = 0;
-                responseLabel.anchorY = 1;
-                responseLabel.textAlign = cc.TEXT_ALIGNMENT_LEFT;
-
-                responseLabel.x = winSize.width / 10 * 3;
-                responseLabel.y = winSize.height / 2;
-                statusPostLabel.setString("Status: Got POST response! " + httpStatus);
-            }
-        };
         xhr.send("plain text message");
+
+        streamXHREventsToLabel(xhr, statusPostLabel, responseLabel, "POST");
```

Comment:
the `streamXHREventsToLabel` should be used before `send`, otherwise `onloadstart` will not be triggered. right?